tests/src/FunctionalJavascript/Views/ActivityFullcalendarViewTest.php

```diff
@@ -0,0 +1,143 @@
+
+namespace Drupal\Tests\civicrm_entity\FunctionalJavascript\Views;
+
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\FunctionalJavascriptTests\DrupalSelenium2Driver;
+use Drupal\Tests\civicrm_entity\FunctionalJavascript\CivicrmEntityTestBase;
+
+/**
+ * @requires module fullcalendar_view
+ */
+final class ActivityFullcalendarViewTest extends CivicrmEntityTestBase {
+
+  /**
+   * {@inheritdoc}
+   *
+   * @todo fix config schema from this module for Views.
+   */
+  protected $strictConfigSchema = FALSE;
+
+  /**
+   * The test activity.
+   *
+   * @var mixed
+   */
+  private $activityId;
+
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'civicrm_entity_fullcalendar_test',
+  ];
+
+
+
+  /**
+   * {@inheritdoc}
+   *
+   * @todo needs more data.
+   */
+  public function setUp(): void {
+    parent::setUp();
+
+    $admin_user = $this->createUser([
+      'access content',
+      'administer civicrm entity',
+    ]);
+    $this->drupalLogin($admin_user);
+    $this->enableCivicrmEntityTypes(['civicrm_activity', 'civicrm_contact']);
+
+    $civicrm_api = $this->container->get('civicrm_entity.api');
+    $result = $civicrm_api->save('Contact', [
+      'contact_type' => 'Individual',
+      'first_name' => 'Johnny',
+      'last_name' => 'Appleseed',
+      'email' => 'johnny@example.com',
+    ]);
+    $contact_id = $result['id'];
+    $user = $this->createUser([], 'johnny');
+
+    // Verify the user and contact linked.
+    $fetched_contact_id = \CRM_Core_BAO_UFMatch::getContactId($user->id());
+    self::assertEquals($contact_id, $fetched_contact_id);
+
+    $date = new DrupalDateTime('now');
+    $result = $civicrm_api->save('Activity', [
+      'source_contact_id' => $contact_id,
+      'activity_type_id' => 'Meeting',
+      'subject' => 'Meeting about new seeds',
+      'activity_date_time' => $date->format('Y-m-d') . ' 12:00:00',
+      'status_id' => 2,
+      'priority_id' => 1,
+      'duration' => 120,
+      'location' => 'Pennsylvania',
+      'details' => 'We need to find more seeds!',
+    ]);
+    $this->activityId = $result['id'];
+
+    \Drupal::configFactory()
+      ->getEditable('views.settings')
+      ->set('ui.always_live_preview', FALSE)
+      ->set('ui.show.advanced_column', TRUE)
+      ->save();
+  }
+
+  public function testFullcalendarDisplay() {
+    $this->drupalGet('/activity-fullcalendar');
+    $this->createScreenshot('../calendar.png');
+    $fullcalendar = $this->assertSession()->elementExists('css', '.js-drupal-fullcalendar');
+    $this->assertSession()->elementExists('css', '.fc-event-container', $fullcalendar);
+    $this->assertSession()->elementTextContains('css', '.fc-event-container .fc-time', '12:00 pm');
+    $this->assertSession()->elementTextContains('css', '.fc-event-container .fc-title', 'Meeting about new seeds');
+
+    // @todo The dialog opened is extremely difficult to target.
+    $this->click('.fc-event-container .fc-event');
```

Comment:
This needs investigation. The dialog opened is very difficult to target. It renders an iframe, so maybe we can toggle into it to assert the output (verify username displayed from relationship)